### PR TITLE
include some fields from context in identify and track

### PR DESCRIPTION
**Pull request by dontgitit**

### CODE REVIEW

**Comment by f2prateek** [comment]
I don't think traits is the right place to set these fields.

**Comment by dontgitit** [comment]
Actually going to add a bit more. Don't merge yet.

**Comment by dontgitit** [comment]
Added some more stuff. Can you guys take a look? Thanks!

**Comment by f2prateek** [comment]
Please squash your commits into one.

File: lib/mapper.js [hunk 1]
```diff
@@ -10,19 +10,50 @@ var unixTime = require('unix-time');
 var del = require('obj-case').del;
 var reject = require('reject');
 
+function copyFields(fromObj, toObj, fields) {
```
Comment (f2prateek) [inline lib/mapper.js]: Instead of this you can use the require external packages like extend and pick `extend(toObj, pick(fromObj, fields))`

Comment (dontgitit) [inline lib/mapper.js]: :+1: 

I found this, is it the right library to use/include? https://www.npmjs.com/package/utils-extend

**Comment by dontgitit** [comment]
Switch to pick+extend and rebased

File: lib/mapper.js (hunk 1 above)
Comment (f2prateek) [inline lib/mapper.js]: I'd prefer if we used https://github.com/ndhoule/pick and https://github.com/ndhoule/extend but it's not a deal breaker.

File: lib/mapper.js [hunk 2]
```diff
@@ -39,12 +60,19 @@ exports.identify = function(identify) {
  */
 
 exports.track = function(track) {
+  var properties = traverse(clone(track.properties()));
```
Comment (f2prateek) [inline lib/mapper.js]: Is it necessary to clone these?

Comment (dontgitit) [inline lib/mapper.js]: Not sure. I copied this from somewhere. What does it do?

File: lib/mapper.js (hunk 1 above)
Comment (dontgitit) [inline lib/mapper.js]: switched to those

File: lib/mapper.js (hunk 2 above)
Comment (f2prateek) [inline lib/mapper.js]: The traverse function cleans up dates (we want this) and the clone function makes a deep copy of the object (we don't need this).

Comment (dontgitit) [inline lib/mapper.js]: Got it. Yeah, don't think we need ```clone```. Removed it from code + package.json

**Comment by dontgitit** [comment]
Thanks for the helpful comments and quick turnaround!

Any chance you know when this will be deployed?

**Pull request merged**